src/lib/core/trigger.cc

```diff
@@ -50,3 +73,82 @@ trigger_run_reverse(struct rlist *list, void *event)
 			return -1;
 	return 0;
 }
+
+void
+trigger_fiber_run(struct rlist *list, void *event, double timeout)
+{
+	struct trigger *trigger, *tmp;
+	unsigned trigger_count = 0;
+	struct region *region = &fiber()->gc;
+	size_t used = region_used(region);
+	bool expired = false;
+	struct ev_timer timer;
+	ev_timer_init(&timer, trigger_fiber_run_timeout, timeout, 0);
+	timer.data = &expired;
+	ev_timer_start(loop(), &timer);
+
+	/*
+	 * Calculating the total number of triggers.
+	 */
+	rlist_foreach_entry(trigger, list, link)
+		trigger_count++;
+
+	struct fiber **fibers = (struct fiber **)
+		region_alloc(region, trigger_count * sizeof(struct fiber *));
+	if (fibers == NULL) {
+		say_error("Failed to allocate %lu bytes in region_alloc "
```

Comment:
Not sure that we should attempt to run triggers on oom. I'd better log error and returned from function.

Reply:
The same happens if we failed to create new fiber below, so doubt relates to that chunk as well.

Reply:
fix